libs/execution_context/ebpf_program.c

```diff
@@ -550,3 +573,29 @@ ebpf_program_free_program_info(_In_opt_ _Post_invalid_ ebpf_program_info_t* prog
         ebpf_free(program_info);
     }
 }
+
+void
+ebpf_program_attach_link(_Inout_ ebpf_program_t* program, _Inout_ ebpf_link_t* link)
+{
+    // Acquire "attach" reference on the link object.
+    ebpf_object_acquire_reference((ebpf_object_t*)link);
+
+    // Insert the link in the attach list.
+    ebpf_lock_state_t state;
+    state = ebpf_lock_lock(&program->links_lock);
+    ebpf_list_insert_tail(&program->links, &((ebpf_object_t*)link)->list_entry);
```

Comment:
program should not ideally access fields of ebpf_link_t. Pass the list entry instead of the link object.

Reply:
oh never mind.